activerecord/lib/active_record/connection_adapters/abstract/schema_statements.rb

```diff
@@ -671,6 +671,29 @@ def remove_reference(table_name, ref_name, options = {})
       def foreign_keys(table_name)
         raise NotImplementedError, "foreign_keys is not implemented"
       end
+      
+      # Checks to see if a foreign key exists on a table for a given foreign key definition.
+      #
+      #   # Check a foreign key exists
+      #   foreign_key_exists?(:accounts, :branches)
+      #
+      #   # Check a foreign key on specified column exists
+      #   foreign_key_exists?(:accounts, column: :owner_id)
+      #
+      #   # Check a foreign key with a custom name exists
+      #   foreign_key_exists?(:accounts, name: "special_fk_name")
+      #
+      def foreign_key_exists?(from_table, options_or_to_table = {})
+        return unless supports_foreign_keys?
+        
+        if options_or_to_table.is_a?(Hash)
+          options = options_or_to_table
+        else
+          options = { column: foreign_key_column_for(options_or_to_table) }
+        end
```

Comment:
I think it'd be good to have the method to access the ForeignKeyDefinition based on the user arguments. There are multiple ways how you can specify a key:
1. `from_table` + `to_table`
2. `from_table` + `column`
3. `from_table` + `name`

Would be nice to have that bundled up so that higher level methods don't have to bother with these details.